### PR TITLE
Log support

### LINKED ISSUE #190: Add data-vis repo link in readme / docs
Add the https://github.com/predict-idlab/ts-datapoint-selection-vis repo somewhere in the README, as it provides more context and information on the performance of data point selection algorithms for various use-cases.

### PR DESCRIPTION
**Pull request by jonasvdd**
Partially addresses #206 
Closes #190

Specifically, log-axis their zoom functionality will be supported.

TODO:
- [ ] tests for log axis and update sizes

### CODE REVIEW

**Comment by jonasvdd** [comment]
Code ⬇️  implementation which projects LTTB its buckets to equidistant log buckets.
```python
import numpy as np
from plotly_resampler.aggregation.aggregation_interface import DataPointSelector
from typing import Union


class LogLTTB(DataPointSelector):
    @staticmethod
    def _argmax_area(prev_x, prev_y, avg_next_x, avg_next_y, x_bucket, y_bucket) -> int:
        """Vectorized triangular area argmax computation.

        Parameters
        ----------
        prev_x : float
            The previous selected point is x value.
        prev_y : float
            The previous selected point its y value.
        avg_next_x : float
            The x mean of the next bucket
        avg_next_y : float
            The y mean of the next bucket
        x_bucket : np.ndarray
            All x values in the bucket
        y_bucket : np.ndarray
            All y values in the bucket

        Returns
        -------
        int
            The index of the point with the largest triangular area.
        """
        return np.abs(
            x_bucket * (prev_y - avg_next_y)
            + y_bucket * (avg_next_x - prev_x)
            + (prev_x * avg_next_y - avg_next_x * prev_y)
        ).argmax()

    def _arg_downsample(
        self, x: Union[np.ndarray, None], y: np.ndarray, n_out: int, **kwargs
    ) -> np.ndarray:
        """TODO complete docs"""
        # We need a valid x array to determing the x-range
        assert x is not None, "x cannot be None for this downsampler"

        # the log function to use
        lf = np.log1p

        offset = np.unique(
            np.searchsorted(
                x, np.exp(np.linspace(lf(x[0]), lf(x[-1]), n_out + 1)).astype(np.int64)
            )
        )

        # Construct the output array
        sampled_x = np.empty(len(offset) + 1, dtype="int64")
        sampled_x[0] = 0
        sampled_x[-1] = x.shape[0] - 1

        # Convert x & y to int if it is boolean
        if x.dtype == np.bool_:
            x = x.astype(np.int8)
        if y.dtype == np.bool_:
            y = y.astype(np.int8)

        a = 0
        for i in range(len(offset) - 2):
            a = (
                self._argmax_area(
                    prev_x=x[a],
                    prev_y=y[a],
                    avg_next_x=np.mean(x[offset[i + 1] : offset[i + 2]]),
                    avg_next_y=y[offset[i + 1] : offset[i + 2]].mean(),
                    x_bucket=x[offset[i] : offset[i + 1]],
                    y_bucket=y[offset[i] : offset[i + 1]],
                )
                + offset[i]
            )
            sampled_x[i + 1] = a

        # ------------ EDGE CASE ------------
        # next-average of last bucket = last point
        sampled_x[-2] = (
            self._argmax_area(
                prev_x=x[a],
                prev_y=y[a],
                avg_next_x=x[-1],  # last point
                avg_next_y=y[-1],
                x_bucket=x[offset[-2] : offset[-1]],
                y_bucket=y[offset[-2] : offset[-1]],
            )
            + offset[-2]
        )
        return sampled_x
```

Code ⬇️ example which creates a log plot:

```python
from plotly_resampler.aggregation import NoGapHandler

n = 100_000
y = np.sin(np.arange(n) / 2_000) + np.random.randn(n) / 10

fr = FigureResampler(False)
fr.add_trace(
    go.Scattergl(mode="lines+markers", marker_color=np.abs(y) / np.max(np.abs(y))),
    hf_x=np.arange(n),
    hf_y=y,
    downsampler=LogLTTB(),
    gap_handler=NoGapHandler(),
)
fr.update_xaxes(type="log")
fr.update_layout(template='plotly_white', title='log axis demo')
```
image

File: tests/test_figure_resampler.py [hunk 1]
```diff
@@ -564,7 +597,89 @@ def test_multiple_timezones():
...
+    # add via scatter kwargs
+    fr = FigureResampler()
+    fr.add_trace(go.Scatter(x=df.timestamp, y=df.value))
+    output = fr.construct_update_data(
+        {
+            "xaxis.range[0]": "2020-01-01 00:00:00",
+            "xaxis.range[1]": "2020-01-01 00:00:20",
+        }
+    )
+    assert len(output) == 2
```
Comment (jvdd) [inline tests/test_figure_resampler.py:682]: Here we observe the duplicate tests.. :exploding_head:

File: plotly_resampler/aggregation/plotly_aggregator_parser.py [hunk 2]
```diff
@@ -60,6 +60,10 @@ def get_start_end_indices(hf_trace_data, start, end) -> Tuple[int, int]:
         start = hf_trace_data["x"][0] if start is None else start
         end = hf_trace_data["x"][-1] if end is None else end
 
+        # NOTE: we must verify this before check if the x is a range-index
+        if axis_type == "log":
+            start, end = 10**start, 10**end
```
[the remark this answers is not in the document]
Comment (jonasvdd) [reply plotly_resampler/aggregation/plotly_aggregator_parser.py:65]: log axis is tested

File: tests/test_figure_resampler.py [hunk 3]
```diff
@@ -239,6 +239,32 @@ def test_box_histogram(float_series):
     )
 
 
+def test_log_axis():
```
Comment (jonasvdd) [inline tests/test_figure_resampler.py:242]: also perform a range index check

**Comment by jvdd** [comment]
LGTM!

**Pull request merged**